src/EditorFeatures/Test/Completion/AbstractCompletionProviderTests.cs

```diff
@@ -45,10 +45,10 @@ public override void Dispose()
             base.Dispose();
         }
 
-        protected static bool CanUseSpeculativeSemanticModel(Document document, int position)
+        protected static async Task<bool> CanUseSpeculativeSemanticModelAsync(Document document, int position)
         {
             var service = document.Project.LanguageServices.GetService<ISyntaxFactsService>();
-            var node = document.GetSyntaxRootAsync().Result.FindToken(position).Parent;
+            var node = (await document.GetSyntaxRootAsync()).FindToken(position).Parent;
```

Comment:
This seems like it would be a great candidate for a refactoring, but I imagine it would get nasty fixing the upstream call sites.